paparazzi/src/main/java/app/cash/paparazzi/internal/validation/MainDispatcherValidator.kt

```diff
@@ -0,0 +1,55 @@
+package app.cash.paparazzi.internal.validation
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainCoroutineDispatcher
+
+/**
+ * Checks if the [MainCoroutineDispatcher] within [Dispatchers.Main] has
+ * been tampered with in a Paparazzi test.
+ *
+ * By default, Paparazzi's [MainCoroutineDispatcher] uses a `HandlerContext.`
+ *
+ * If someone uses `Dispatchers.setMain(...)` within a Paparazzi test, there is a chance
+ * that they make the test pass, but not really execute some of the scheduled coroutines within
+ * the test dispatcher. This in turn can pile up in large modules with lots of snapshot tests, and
+ * eventually can cause `java.lang.OutOfMemoryError.`
+ *
+ * Setting the `Dispatchers.Main` to a `TestDispatcher` can also lead to flaky tests if the developer
+ * doesn't properly call `advanceUntilIdle()`.
+ *
+ * Finally, Paparazzi already has access to its own `AndroidUiDispatcher,` so there
+ * aren't many benefits to use `Dispatchers.setMain()` within Paparazzi tests.
```

Comment:
Thank you for the contribution ❤️ 

I'm still trying to understand the use cases here though, namely:
* In what case would someone deliberately modify Dispatchers.Main?
* Even so, is it appropriate for this library to be the watchdog for this potential scenario versus a custom test rule in your codebase that would check for this?

Additionally, I'm not understanding where preventing Dispatchers.Main from being set prevents OOMEs from occurring.

Reply:
hmm, in that case I feel like the responsibility of managing main dispatcher is less something the `Paparazzi` should manage and more on the plugin consumers. Curious on your thoughts here @jrodbx